backend/src/api/game/game_response.rs

```diff
@@ -0,0 +1,86 @@
+use hive_lib::{
+    bug::Bug, game_status::GameStatus, game_type::GameType, piece::Piece, position::Position,
+};
+use serde::Serialize;
+use serde_with::serde_as;
+use std::collections::HashMap;
+
+#[serde_as]
+#[serde_with::skip_serializing_none]
+#[derive(Serialize)]
+pub struct GameResponse {
+    game_id: u64,
+    turn: usize,
+    game_status: GameStatus,
+    game_type: GameType,
+    tournament: bool,
+    // this should just serialize users later on
+    white_uid: String,
+    white_username: String,
+    white_elo: f64,
+
+    // this should just serialize users later on
+    black_uid: String,
+    black_username: String,
+    black_elo: f64,
+
+    #[serde_as(as = "Vec<((_, _), _)>")]
+    moves: HashMap<(String, Position), Vec<Position>>,
+    spawns: Vec<Position>,
+    reserve: HashMap<Bug, i8>,
+    history: Vec<(String, String)>,
+}
+
+impl GameResponse {
+    // TODO we don't need to pass everything in once we have a couple of good helpers :)
+    pub fn new(
+        game_id: u64,
+        turn: usize,
+        game_status: GameStatus,
+        game_type: GameType,
+        tournament: bool,
+        moves: HashMap<(Piece, Position), Vec<Position>>,
+        spawns: Vec<Position>,
+        reserve: HashMap<Bug, i8>,
+        history: Vec<(String, String)>,
+    ) -> Self {
+        let ((white_uid, white_username, white_elo), (black_uid, black_username, black_elo)) =
+            GameResponse::get_user_from_game_id(game_id);
+
+        Self {
+            game_id,
+            game_status,
+            game_type,
+            tournament,
+            white_uid,
+            white_username,
+            white_elo,
+            black_uid,
+            black_username,
+            black_elo,
+            turn,
+            moves: GameResponse::moves_as_string(moves),
+            spawns,
+            reserve,
+            history,
+        }
+    }
+
+    fn get_user_from_game_id(_game_id: u64) -> ((String, String, f64), (String, String, f64)) {
```

Comment:
if we're sketching out what the real API would be, it'd be better to return a `User` here